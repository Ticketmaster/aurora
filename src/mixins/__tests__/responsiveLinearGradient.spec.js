import { responsiveLinearGradient } from "../";

describe("responsiveLinearGradient", () => {
  it("should return a background-image style rule when deg and stops args are passed", () => {
    expect(
      responsiveLinearGradient({
        deg: { small: "256deg" },
        stops: ["rgb(0, 45, 161)", "rgb(2, 108, 223) 55%", "rgb(0, 45, 161)"]
      })
    ).toMatchSnapshot();
  });

  it("should return a background-image style rule when deg and stops args are passed", () => {
    expect(
      responsiveLinearGradient({
        deg: { small: "256deg" },
        stops: ["rgb(0, 45, 161)", "rgb(2, 108, 223) 55%", "rgb(0, 45, 161)"]
      })
    ).toMatchSnapshot();
  });

  it("should return a background-image style rule for multiple device sizes when deg and stops args are passed", () => {
    expect(
      responsiveLinearGradient({
        deg: { small: "256deg", medium: "240deg", large: "220deg" },
        stops: ["rgb(0, 45, 161)", "rgb(2, 108, 223) 55%", "rgb(0, 45, 161)"]
      })
    ).toMatchSnapshot();
  });

  it("should return no rule when only a deg arg is passed", () => {
    expect(
      responsiveLinearGradient({
        deg: { small: "256deg" }
      })
    ).toMatchSnapshot();
  });

  it("should return no rule when only a stops arg is passed", () => {
    expect(
      responsiveLinearGradient({
        stops: ["rgb(0, 45, 161)", "rgb(2, 108, 223) 55%", "rgb(0, 45, 161)"]
      })
    ).toMatchSnapshot();
  });
});
