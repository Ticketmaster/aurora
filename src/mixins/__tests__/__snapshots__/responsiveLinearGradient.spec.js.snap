// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsiveLinearGradient should return a background-image style rule for multiple device sizes when deg and stops args are passed 1`] = `
Array [
  "
    ",
  "
    background-image: linear-gradient(",
  "256deg",
  ", ",
  "rgb(0, 45, 161), rgb(2, 108, 223) 55%, rgb(0, 45, 161)",
  ");
  ",
  ";
    ",
  "
  @media screen and ",
  "(min-width: 768px)",
  " {
    ",
  "
    background-image: linear-gradient(",
  "240deg",
  ", ",
  "rgb(0, 45, 161), rgb(2, 108, 223) 55%, rgb(0, 45, 161)",
  ");
  ",
  ";
  }
",
  " ",
  "
  @media screen and ",
  "(min-width: 1024px)",
  " {
    ",
  "
    background-image: linear-gradient(",
  "220deg",
  ", ",
  "rgb(0, 45, 161), rgb(2, 108, 223) 55%, rgb(0, 45, 161)",
  ");
  ",
  ";
  }
",
  ";
  ",
]
`;

exports[`responsiveLinearGradient should return a background-image style rule when deg and stops args are passed 1`] = `
Array [
  "
    ",
  "
    background-image: linear-gradient(",
  "256deg",
  ", ",
  "rgb(0, 45, 161), rgb(2, 108, 223) 55%, rgb(0, 45, 161)",
  ");
  ",
  ";
    ",
  " ",
  ";
  ",
]
`;

exports[`responsiveLinearGradient should return a background-image style rule when deg and stops args are passed 2`] = `
Array [
  "
    ",
  "
    background-image: linear-gradient(",
  "256deg",
  ", ",
  "rgb(0, 45, 161), rgb(2, 108, 223) 55%, rgb(0, 45, 161)",
  ");
  ",
  ";
    ",
  " ",
  ";
  ",
]
`;

exports[`responsiveLinearGradient should return no rule when only a deg arg is passed 1`] = `undefined`;

exports[`responsiveLinearGradient should return no rule when only a stops arg is passed 1`] = `undefined`;
